test: Intermittent issue in p2p_compactblocks_hb.py"  in relay_block_through                                        assert_equal(status_to, status_from)  AssertionError: not([True, False, True, True] == [True, True, True, True])

Logs: https://drahtbot.space/temp_scratch/p2p_compactblocks_hb_170.tar.zstd

```
 test  2024-04-10T02:55:04.098000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/ci_container_base/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 132, in main
                                       self.run_test()
                                     File "/ci_container_base/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/p2p_compactblocks_hb.py", line 79, in run_test
                                       status = self.relay_block_through(nodeid)
                                     File "/ci_container_base/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/p2p_compactblocks_hb.py", line 37, in relay_block_through
                                       assert_equal(status_to, status_from)
                                     File "/ci_container_base/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/util.py", line 74, in assert_equal
                                       raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
                                   AssertionError: not([True, False, True, True] == [True, True, True, True])

--- a/test/functional/p2p_compactblocks_hb.py
+++ b/test/functional/p2p_compactblocks_hb.py
@@ -32,10 +32,15 @@ class CompactBlocksConnectionTest(BitcoinTestFramework):
         self.connect_nodes(peer, 0)
         self.generate(self.nodes[0], 1)
         self.disconnect_nodes(peer, 0)
-        status_to = [self.peer_info(1, i)['bip152_hb_to'] for i in range(2, 6)]
-        status_from = [self.peer_info(i, 1)['bip152_hb_from'] for i in range(2, 6)]
-        assert_equal(status_to, status_from)
-        return status_to
+
+        def status_to():
+            return [self.peer_info(1, i)['bip152_hb_to'] for i in range(2, 6)]
+
+        def status_from():
+            return [self.peer_info(i, 1)['bip152_hb_from'] for i in range(2, 6)]
+
+        self.wait_until(lambda: status_to() == status_from())
+        return status_to()
 
     def run_test(self):
         self.log.info("Testing reserved high-bandwidth mode slot for outbound peer...")

